proof/refine/AARCH64/ArchInvsDefs_H.thy

```diff
@@ -39,123 +36,20 @@ end
 
 end
 
-\<comment> \<open>---------------------------------------------------------------------------\<close>
-
-section "Invariants on Executable Spec"
-
-context begin interpretation Arch .
-
-definition ps_clear :: "obj_ref \<Rightarrow> nat \<Rightarrow> kernel_state \<Rightarrow> bool" where
-  "ps_clear p n s \<equiv> (mask_range p n - {p}) \<inter> dom (ksPSpace s) = {}"
-
-definition pspace_no_overlap' :: "obj_ref \<Rightarrow> nat \<Rightarrow> kernel_state \<Rightarrow> bool" where
-  "pspace_no_overlap' ptr bits \<equiv>
-     \<lambda>s. \<forall>x ko. ksPSpace s x = Some ko \<longrightarrow>
-                (mask_range x (objBitsKO ko)) \<inter> {ptr .. (ptr && ~~ mask bits) + mask bits} = {}"
-
-definition ko_wp_at' :: "(kernel_object \<Rightarrow> bool) \<Rightarrow> obj_ref \<Rightarrow> kernel_state \<Rightarrow> bool" where
-  "ko_wp_at' P p s \<equiv> \<exists>ko. ksPSpace s p = Some ko \<and> is_aligned p (objBitsKO ko) \<and> P ko \<and>
-                           ps_clear p (objBitsKO ko) s"
-
-definition obj_at' :: "('a::pspace_storable \<Rightarrow> bool) \<Rightarrow> machine_word \<Rightarrow> kernel_state \<Rightarrow> bool" where
-  obj_at'_real_def:
-  "obj_at' P p s \<equiv> ko_wp_at' (\<lambda>ko. \<exists>obj. projectKO_opt ko = Some obj \<and> P obj) p s"
-
-definition typ_at' :: "kernel_object_type \<Rightarrow> machine_word \<Rightarrow> kernel_state \<Rightarrow> bool" where
-  "typ_at' T \<equiv> ko_wp_at' (\<lambda>ko. koTypeOf ko = T)"
-
-abbreviation ep_at' :: "obj_ref \<Rightarrow> kernel_state \<Rightarrow> bool" where
-  "ep_at' \<equiv> obj_at' ((\<lambda>x. True) :: endpoint \<Rightarrow> bool)"
-
-abbreviation ntfn_at' :: "obj_ref \<Rightarrow> kernel_state \<Rightarrow> bool" where
-  "ntfn_at' \<equiv> obj_at' ((\<lambda>x. True) :: notification \<Rightarrow> bool)"
-
-abbreviation tcb_at' :: "obj_ref \<Rightarrow> kernel_state \<Rightarrow> bool" where
-  "tcb_at' \<equiv> obj_at' ((\<lambda>x. True) :: tcb \<Rightarrow> bool)"
-
-abbreviation real_cte_at' :: "obj_ref \<Rightarrow> kernel_state \<Rightarrow> bool" where
-  "real_cte_at' \<equiv> obj_at' ((\<lambda>x. True) :: cte \<Rightarrow> bool)"
+context Arch begin arch_global_naming
 
-abbreviation ko_at' :: "'a::pspace_storable \<Rightarrow> obj_ref \<Rightarrow> kernel_state \<Rightarrow> bool" where
-  "ko_at' v \<equiv> obj_at' (\<lambda>k. k = v)"
+declare lookupPTSlotFromLevel.simps[simp del]
+declare lookupPTFromLevel.simps[simp del]
 
-abbreviation
+abbreviation vcpu_at' :: "obj_ref \<Rightarrow> kernel_state \<Rightarrow> bool" where
   "vcpu_at' \<equiv> typ_at' (ArchT VCPUT)"
 
 abbreviation pte_at' :: "obj_ref \<Rightarrow> kernel_state \<Rightarrow> bool" where
   "pte_at' \<equiv> typ_at' (ArchT PTET)"
 
-end
-
-record itcb' =
```

Comment:
Huh, not relevant for this PR but I'm mildly surprised that we don't have a Refine equivalent to AInvs' `iarch_tcb`. In AInvs that lets us reuse the `pred_tcb_at` machinery for vcpus as well.

Reply:
Then no one did such a thing in rt either?

Reply:
Nope, rt so far has only been for architectures without hypervisor support, so there's never been a need for `pred_tcb_at` of an `arch_tcb` property.

Reply:
This strikes fear into my heart